whole_body_kinematic_controller:
  type: pal_wbc/WholeBodyControlKinematicDiffDriveController
  
  useFT: False
  useIMU: False
  dt: 0.005
  floating_base: XY_Yaw
  formulation: velocity
  stack_configuration: tiago_mobile_base_stack
  solver_type: QpReductionuEqualitiesQuadprogHeapAllocation
  real_hardware: False
  
  left_wheel_joint_name: wheel_left_joint
  right_wheel_joint_name: wheel_right_joint
  use_wheel_velocity_control: false
  wheel_separation: 0.4044
  wheel_radius: 0.0985
  max_linear_velocity_limit: 0.2
  max_angular_velocity_limit: 0.2
  max_wheel_velocity_limit:  0.2
  mobile_base_controller_topic: /mobile_base_controller/cmd_vel

  use_home_posture: True

  robot_model_chains:
    [wrist_ft_link,
     xtion_optical_frame]

  position_joints:        
    - arm_1_joint
    - arm_2_joint
    - arm_3_joint
    - arm_4_joint
    - arm_5_joint
    - arm_6_joint
    - arm_7_joint
    - head_1_joint
    - head_2_joint
    - torso_lift_joint

  reference:
    arm_1_joint: 0.11
    arm_2_joint: -0.51
    arm_3_joint: -3.22
    arm_4_joint: 2.0
    arm_5_joint: 1.91
    arm_6_joint: 0.36
    arm_7_joint: 0.3
    head_1_joint: 0.0 
    head_2_joint: 0.0
    torso_lift_joint: 0.14